tests/testsuite/cargo_command.rs

```diff
@@ -341,6 +341,58 @@ fn cargo_subcommand_env() {
         .run();
 }
 
+#[cfg(unix)]
```

Comment:
I would prefer to not have this be Unix-only if possible. I think something like the following should be sufficient for creating some executables to run.

```rust
    let p = project()
        .at("cargo-foo")
        .file("Cargo.toml", &basic_manifest("cargo-foo", "1.0.0"))
        .file("src/bin/cargo-foo.rs", r#"fn main() { println!("INSIDE"); }"#)
        .file("src/bin/cargo-foo2.rs", r#"fn main() { println!("OUTSIDE"); }"#)
        .build();
    p.cargo("build").run();
    let cargo_bin_dir = paths::home().join(".cargo/bin");
    cargo_bin_dir.mkdir_p();
    let root_bin_dir = paths::root().join("bin");
    root_bin_dir.mkdir_p();
    let exe_name = format!("cargo-foo{}", env::consts::EXE_SUFFIX);
    fs::rename(p.bin("cargo-foo"), cargo_bin_dir.join(&exe_name)).unwrap();
    fs::rename(p.bin("cargo-foo2"), root_bin_dir.join(&exe_name)).unwrap();
```

and use join_paths below.  This example also doesn't require setting `CARGO_HOME`.

This will fail on restricted systems that don't have bash, for example.

Reply:
Done